lib/routes/ups/track.ts

```diff
@@ -0,0 +1,107 @@
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['ups.com/track?loc=en_US&tracknum=:trackingNumber'],
+            target: '/ups/track/:trackingNumber',
+        },
+    ],
+    name: 'UPS Tracking',
+    maintainers: ['Aquabet'],
+    handler,
+};
+
+async function handler(ctx) {
+    const { trackingNumber } = ctx.req.param();
+    const url = `https://www.ups.com/track?loc=en_US&tracknum=${trackingNumber}`;
+
+    const browser = await puppeteer();
+    const page = await browser.newPage();
+
+    await page.setRequestInterception(true);
+
+    // skip loading images, stylesheets, and fonts
+    page.on('request', (request) => {
+        if (['image', 'stylesheet', 'font'].includes(request.resourceType())) {
+            request.abort();
+        } else {
+            request.continue();
+        }
+    });
+
+    await page.goto(url, { waitUntil: 'domcontentloaded' });
+
+    await page.waitForSelector('tr[id^="stApp_ShpmtProg_LVP_progress_row_"]', {
+        timeout: 30000,
+    });
+
+    const content = await page.content();
+    browser.close();
+    const $ = load(content);
+
+    // Extract tracking events
+    const items = $('tr[id^="stApp_ShpmtProg_LVP_progress_row_"]')
+        .toArray()
+        .flatMap((el) => {
+            const $el = $(el);
+
+            // Extract status, location, and datetime
+            const status = $el
+                .find(`td[id^="stApp_ShpmtProg_LVP_milestone_nameKey_"]`)
+                .contents()
+                .filter(function () {
+                    return this.type === 'text' && this.data.trim() !== '';
+                })
+                .toArray()
+                .map((element) => element.data)
+                .join('')
+                .trim();
+
+            const location = $el.find(`span[id^="stApp_milestoneLocation"]`).text().trim();
+            const dateTimeText = $el.find(`span[id^="stApp_milestoneDateTime"]`).text().trim();
+
+            if (!dateTimeText) {
+                return [];
+            }
+
+            const cleanedDateTimeText = dateTimeText.normalize('NFKC').replaceAll(/\s+/g, ' ').replaceAll('A.M.', 'AM').replaceAll('P.M.', 'PM').trim();
+
+            // Separate date and time
+            const [date, time] = cleanedDateTimeText.split(', ');
+            const formattedDateTime = `${date} ${time}`;
+
+            return {
+                title: `${status}: ${location}`,
+                link: url,
+                description: `
+                    Status: ${status} <br>
+                    Location: ${location} <br>
+                    Date and Time: ${date} ${time}
+                `.trim(),
+                pubDate: parseDate(formattedDateTime, 'MM/DD/YYYY h:mm A'),
+            };
```

Comment:
Please create a guid as `link` is not unique

Reply:
UPS does not provide a separate page for each tracking record. That's why I'm linking each record to the tracking page.

Reply:
Without provide a guid for feed item, RSSHub will use `link` as guid. Since `guid` is the same across all items, some RSS readers will show only one entry.